pkg/api/kptfile/v1alpha2/validation.go

```diff
@@ -147,13 +143,50 @@ func validateFnConfigPath(p string) error {
 	return nil
 }
 
+func GetValidatedFnConfigFromPath(pkgPath types.UniquePath, configPath string) (*yaml.RNode, error) {
+	path := filepath.Join(string(pkgPath), configPath)
+	file, err := os.Open(path)
+	if err != nil {
+		return nil, fmt.Errorf("functionConfig must exist in the current package")
+	}
+	reader := kio.ByteReader{Reader: file}
+	nodes, err := reader.Read()
+	if err != nil {
+		return nil, fmt.Errorf("failed to read function config '%s': %w", configPath, err)
+	}
+	if len(nodes) > 1 {
+		return nil, fmt.Errorf("function config file '%s' must not contain more than 1 config, got %d", configPath, len(nodes))
+	}
+	if err := IsKrm(nodes[0]); err != nil {
+		return nil, fmt.Errorf("invalid function config file '%s': %s", configPath, err.Error())
```

Comment:
we can rearrange the wording to read better:
`functionConfig %q: %s`, configPath, err
It will read: `functionConfig "/tmp/something-path": resource must have `kind``

Reply:
fixed these error messages, thank you!